src/main/java/seedu/address/logic/commands/RemarkCommand.java

```diff
@@ -0,0 +1,103 @@
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Edits the remark for a person specified in the INDEX
+ */
+public class RemarkCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "remark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the remark of the person identified "
+        + "by the index number used in the last person listing. "
+        + "Existing remark will be overwritten by the input.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + PREFIX_REMARK + "[REMARK]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_REMARK + "Likes to swim.";
+
+    private static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Person: %1$s";
+    private static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed remark from Person: %1$s";
+    private static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final Index index;
+    private final Remark remark;
+
+    /**
+    * @param index of the person in the filtered person list to edit the remark
+    * @param remark of the person
+    */
+    public RemarkCommand(Index index, Remark remark) {
+        requireNonNull(index);
+        requireNonNull(remark);
+
+        this.index = index;
+        this.remark = remark;
+    }
+
+    @Override
+    protected CommandResult executeUndoableCommand() throws CommandException {
+
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        ReadOnlyPerson personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getAddress(), remark, personToEdit.getTags());
+
+        try {
+            model.updatePerson(personToEdit, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
+        }
+
+        return new CommandResult(generateSuccessMessage(editedPerson));
+    }
+
+    /**
+     * Returns sucess message depends on different scenario such as adding or deleting remarks
+     */
+    private String generateSuccessMessage(ReadOnlyPerson personToEdit) {
+        if (!remark.value.isEmpty()) {
+            return String.format(MESSAGE_ADD_REMARK_SUCCESS, personToEdit);
+        } else {
+            return String.format(MESSAGE_DELETE_REMARK_SUCCESS, personToEdit);
+        }
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Coding style does not match for equals method implementation from other classes.